### ISSUE TITLE
Omnivore is dying... 

### ISSUE THREAD

**Issue by vronizor** [Rnbsov/Omnivore-Telegram-Bot#14]
I don't know if you heard the news, but [Omnivore will shut down](https://blog.omnivore.app/p/details-on-omnivore-shutting-down)... very sad as I used it very satisfactorily every day up to know, and with the great help of your bot :)

I'm looking at alternatives and might be going to Readwise Reader, which allows to send email and subscribe to newsletters using their custom email.

The only sticking point is the Telegram forwarding... Do you know of any trustful bots for Readwise Reader? Would you be interested in coding a similar bot for Readwise?

Just putting it out there — no pressure of course!

Thanks again for all the great service you've offered me so far :)

**Comment by Rnbsov** [comment]
> I don't know if you heard the news, but [Omnivore will shut down](https://blog.omnivore.app/p/details-on-omnivore-shutting-down)... very sad as I used it very satisfactorily every day up to know, and with the great help of your bot :)
> 

No way that's happening! 😭

That's so sad to hear, I was also using this service for a long time!

**Comment by Rnbsov** [comment]
> I'm looking at alternatives and might be going to Readwise Reader, which allows to send email and subscribe to newsletters using their custom email.


For me `Readwise Reader` always seemed to be the best option, but unfortunatelly as a student I couldn't afford it.

**Comment by Rnbsov** [comment]
> The only sticking point is the Telegram forwarding... Do you know of any trustful bots for Readwise Reader? Would you be interested in coding a similar bot for Readwise?

I would actully be happy to create a Telegram bot for `Readwise Reader`! 🚀 I can start from scratch since the source code for the `Omnivore-Telegram-Bot` was a bit messy, and I didn’t have enough time to maintain it properly.

**Comment by Rnbsov** [comment]
> Thanks again for all the great service you've offered me so far :)

You're welcome! 😄👍 

I'm glad to hear that my skills have found some applications and that I've been able to make life a little easier for someone!

**Comment by Rnbsov** [comment]
I don’t think Omnivore will completely die!  Since the source code is open to anyone, I believe there will always be people willing to host it publicly. Plus, you can easily set up your own instance of Omnivore if needed! 🌐

**Comment by Rnbsov** [comment]
Just checked the API of `Readwise Reader` and it seems to be not a big deal of creating the integration 👍

**Comment by vronizor** [comment]
Hi @Rnbsov 

> I would actully be happy to create a Telegram bot for Readwise Reader! 🚀 I can start from scratch since the source code for the Omnivore-Telegram-Bot was a bit messy, and I didn’t have enough time to maintain it properly.

That would be awesome!

> I don’t think Omnivore will completely die! Since the source code is open to anyone, I believe there will always be people willing to host it publicly. Plus, you can easily set up your own instance of Omnivore if needed! 🌐

I was looking into this at the time of the announcement and didn't see much action re people hosting it publicly. Maybe I didn't look hard enough and/or things might have changed since. Setting up my own instance is slightly beyond the effort I'd be happy to make at the moment...

> Just checked the API of Readwise Reader and it seems to be not a big deal of creating the integration 👍

Awesome! Let me know when you have a prototype ready and would be happy to test! You have probably seen this already but just wanted to flag that someone has already done a bot for readwise: https://github.com/ixnet/readwise_telegram_bot. However, it needs to be self hosted because the author is worried about storing people's tokens. How do you feel about this? Do you also think it's a security concern?

**Comment by Rnbsov** [comment]
> Awesome! Let me know when you have a prototype ready and would be happy to test! You have probably seen this already but just wanted to flag that someone has already done a bot for readwise: [ixnet/readwise_telegram_bot](https://github.com/ixnet/readwise_telegram_bot). However, it needs to be self hosted because the author is worried about storing people's tokens. How do you feel about this? Do you also think it's a security concern?

Yeah, the first thing I did was to search for existing implementations. I could use the code base of this repository to make my implementation, and I would probably store the tokens myself so everyone could use the bot without the need to self host it somewhere 

But unfortunately I can't afford `Readwise Reader`, so I'm not sure I could test it. That's the main obstacle currently for me.

**Comment by vronizor** [comment]
> But unfortunately I can't afford Readwise Reader, so I'm not sure I could test it. That's the main obstacle currently for me.

If it's any help, there's a 30 day trial period (which I'm currently enjoying). Otherwise, happy to help out for a couple of months, or test it on my side. Let me know!
